File: apps/sdkwork-webserver-flutter-mobile/packages/sdkwork_webserver_flutter_mobile_core/lib/composition/sdk_inventory.dart
/// SDK client workspaces this root declares.
///
/// Authority: `specs/component.spec.json` `contracts.sdkDependencies`.
///
/// `dartArtifactAvailable` records whether a generated Dart/Flutter variant of
/// that workspace exists today. It is deliberately part of the inventory rather
/// than a comment: neither `sdkwork-drive` nor `sdkwork-deployments` ships a
/// Dart variant today — both are TypeScript-only — and the `deploy_app` catalog
/// port in `sdk/` is unbound for exactly that reason.
///
/// The legacy `sdkwork-webserver-app-sdk` family is deliberately absent: the
/// `deploy_app` entity is owned by `sdkwork-deployments`, so this root no longer
/// composes the webserver-owned application surface
/// (`COMPOSABLE_ARCHITECTURE_SPEC.md` §7: one owner per normalized route).
class WebserverFlutterCoreSdkInventoryEntry {
  const WebserverFlutterCoreSdkInventoryEntry({
    required this.workspace,
    required this.packageName,
    required this.permissionModuleId,
    required this.dartArtifactAvailable,
  });

  final String workspace;
  final String packageName;
  final String permissionModuleId;
  final bool dartArtifactAvailable;
}

const List<WebserverFlutterCoreSdkInventoryEntry>
webserverFlutterCoreSdkInventory = <WebserverFlutterCoreSdkInventoryEntry>[
  WebserverFlutterCoreSdkInventoryEntry(
    workspace: 'sdkwork-drive-app-sdk',
    packageName: 'sdkwork_drive_app_sdk',
    permissionModuleId: 'drive',
    dartArtifactAvailable: false,
  ),
  WebserverFlutterCoreSdkInventoryEntry(
    workspace: 'sdkwork-deployments-app-sdk',
    packageName: 'sdkwork_deployments_app_sdk',
    permissionModuleId: 'deployments',
    dartArtifactAvailable: false,
  ),
];

List<WebserverFlutterCoreSdkInventoryEntry> listWebserverFlutterCoreSdkInventory() =>
    webserverFlutterCoreSdkInventory;

File: apps/sdkwork-webserver-h5/packages/sdkwork-webserver-h5-core/src/composition/sdk-inventory.ts
/**
 * Core-owned SDK inventory. Mirrors
 * `specs/component.spec.json#contracts.sdkClients` and is the single place the
 * H5 application root declares which generated app SDK families it composes.
 *
 * The `deploy_app` entity is owned by `sdkwork-deployments`; this root consumes
 * that authority through `@sdkwork/deployments-app-sdk`. The legacy
 * `@sdkwork/webserver-app-sdk` family is not composed here — it belongs to the
 * webserver app-api surface that the deployments-owned one supersedes
 * (`COMPOSABLE_ARCHITECTURE_SPEC.md` §7: one owner per normalized route).
 */
export function listWebserverH5CoreSdkInventory() {
  return [
    {
      packageName: "@sdkwork/drive-app-sdk",
      authority: "sdkwork-drive-app-api",
      surface: "app-api",
    },
    {
      packageName: "@sdkwork/deployments-app-sdk",
      authority: "sdkwork-deployments-app-api",
      surface: "app-api",
    },
  ] as const;
}

export type WebserverH5CoreSdkInventoryEntry =
  ReturnType<typeof listWebserverH5CoreSdkInventory>[number];
